exchange/exchange.go

```diff
@@ -1102,6 +1089,33 @@ func applyCategoryMapping(ctx context.Context, targeting openrtb_ext.ExtRequestT
 	return res, seatBids, rejections, nil
 }
 
+// findDurationRange returns the element in the array 'a' that is both greater than 'dur' and closest
+// in value to 'dur' unless a value equal to 'dur' is found. Returns an error if all elements in 'a'
+// are less than 'dur'.
+func findDurationRange(dur int, a []int) (int, error) {
+	newDur := dur
+	madeSelection := false
+	var err error
+
+	for i := range a {
+		if dur > a[i] {
+			continue
+		}
+		if dur == a[i] {
+			return a[i], nil
+		}
+		// dur < a[i]
+		if a[i] < newDur || !madeSelection {
+			newDur = a[i]
+			madeSelection = true
+		}
+	}
+	if !madeSelection && len(a) > 0 {
+		err = errors.New("Bid duration exceeds maximum allowed")
```

Comment:
Very minor nitpick: I think error messages are supposed to be entirely lowercase? I could be wrong on this